packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_performance.py

```diff
@@ -0,0 +1,47 @@
+import sys
+import time
+from unittest import TestCase
+import pytest
+import numpy as np
+import plotly.graph_objs as go
+import plotly.io as pio
+from plotly.tests.b64 import b64, _b64
+
+
+def test_performance_b64():
+    rand_arr_1 = np.random.random(100000)
+    rand_arr_2 = np.random.random(100000)
+    raw_arr_1 = rand_arr_1.tolist()
+    raw_arr_2 = rand_arr_2.tolist()
+    b64_arr_1 = b64(rand_arr_1)
+    b64_arr_2 = b64(rand_arr_2)
+
+    # Test the performance of the base64 arrays
+    b64_start = time.time()
+    fig = go.Scatter(x=b64_arr_1, y=b64_arr_2)
+    b64_time_elapsed = time.time() - b64_start
+
+    # Test the performance of the raw arrays
+    raw_start = time.time()
+    fig = go.Scatter(x=raw_arr_1, y=raw_arr_2)
+    raw_time_elapsed = time.time() - raw_start
+
+    # b64 should be faster than raw
+    assert (b64_time_elapsed / raw_time_elapsed) < 0.85
+
+
+def test_size_performance_b64_uint8():
+    rand_arr_1 = np.random.random(100000).astype("uint8")
+    rand_arr_2 = np.random.random(100000).astype("uint8")
```

Comment:
We need to multiply `[0, 1)` random numbers by 256 before casting to `uint8`; otherwise what would get just zeros.
```suggestion
    rand_arr_1 = (np.random.random(100000) * 256).astype("uint8")
    rand_arr_2 = (np.random.random(100000) * 256).astype("uint8")
```

Reply:
Alternative you could use `randint` similar to https://github.com/plotly/plotly.py/pull/4695#issuecomment-2263263589.